src/catnap/mod.rs

```diff
@@ -346,8 +364,29 @@ impl CatnapLibOS {
     }
 
     /// Waits for any operation to complete.
-    pub fn wait_any(&mut self, _qts: &[QToken]) -> (usize, dmtr_qresult_t) {
-        todo!() // TODO: implement this function.
+    pub fn wait_any(&mut self, qts: &[QToken]) -> (usize, dmtr_qresult_t) {
+        #[cfg(feature = "profiler")]
+        timer!("catnap::wait_any");
+        trace!("wait_any(): qts={:?}", qts);
+
+        loop {
+            // Poll first, so as to give a chance of pending operations to complete.
+            self.runtime.poll();
+
+            // Search for any operation that has completed.
+            for (i, &qt) in qts.iter().enumerate() {
+                let handle: SchedulerHandle = self.runtime.get_handle(qt.into()).unwrap();
```

Comment:
Is this unwrap() guaranteed to never fail?

Reply:
That's a catch and it is indeed a bug.  We cannot assume that the supplied `qtoken` is valid at all. I've created issue https://github.com/demikernel/demikernel/issues/71 to keep track of this.

Reply:
Great, thanks.